apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx

```diff
@@ -144,21 +146,38 @@ export default class AnimationPickerBody extends React.Component {
   animationCategoriesRendering() {
     let categories = Object.keys(this.props.libraryManifest.categories || []);
     if (this.props.hideBackgrounds) {
-      categories = categories.filter(category => category !== 'backgrounds');
+      categories = categories.filter(category => category !== BACKGROUNDS);
+    }
+    // Level-specific animations currently have the following categories:
+    // "level_costumes", "backgrounds", "animals".
+    // We want to hide the "animals" category which has only one animation in it.
+    // It will be displayed in the "all" category added below.
+    // There is an "animals" category because a level-specific animation was uploaded with the
+    // "animals" category BEFORE the sprite upload feature dictated that level-specific animations
+    // can only be categorized as "costume" or "background".
+    // WE check if LEVEL_COSTUMES is in the category list below since it implies that these are
+    // level-specific animations (as opposed to library animations) and therefore the need for
+    // the filtering described above.
+    if (categories.includes(LEVEL_COSTUMES)) {
+      categories = categories.filter(
+        category => category === BACKGROUNDS || category === LEVEL_COSTUMES
+      );
```

Comment:
Not blocking on this, since I don't know which way the logic should flow, but would it be better as:
```
category => category !== ANIMALS
```

As is, it'll only include `BACKGROUNDS` or `LEVEL_COSTUMES`. Which is fine, unless there's a chance we'd add another new one that we'd want to include by default.

Keeping it this way would ensure that only `BACKGROUNDS` and `LEVEL_COSTUMES` come through, even if something else is added.

I don't know what the chances of that happening or which way we'd want to fall on it, though, so just raising as a question.

Reply:
Content editors prefer that there are only two categories for level-specific animations - 'BACKGROUNDS' and 'LEVEL_COSTUMES'. If an engineer inadvertently adds a level-specific animation to S3 manually with a different category like 'sports', then we do not want this category to be displayed. Currently, using the [sprite upload feature](https://levelbuilder-studio.code.org/sprites/sprite_upload), level-specific animations can only either 'costumes' or 'backgrounds'.

Reply:
Thanks for the clarification! Then definitely leave as is.